components/org.wso2.carbon.crypto.provider/src/main/java/org/wso2/carbon/crypto/provider/SymmetricKeyInternalCryptoProvider.java

```diff
@@ -153,6 +153,69 @@ public byte[] decrypt(byte[] ciphertext, String algorithm, String javaSecurityAP
         }
     }
 
+    /**
+     * Computes and returns the cleartext of the given ciphertext.
+     *
+     * @param ciphertext                The ciphertext to be decrypted.
+     * @param algorithm                 The encryption / decryption algorithm
+     * @param javaSecurityAPIProvider   The Java Security Provider.
+     * @return The cleartext
+     * @throws CryptoException If something unexpected happens during the decryption operation.
+     */
+    public byte[] decrypt(byte[] ciphertext, String algorithm, String javaSecurityAPIProvider, Object... params)
```

Comment:
we should be able to merge the common logic in above decrypt method and this method. Check for possibilities. That way, we can remove code duplications

Reply:
There are many method overloading with different set of parameters. Here, we are passing an optional object which we can use for other new parameters as well. If the passed params are null, we call the default methods. Otherwise we need to do the other validations, where we are additionally passing the customs parameters as well.